api/middleware.js

```diff
@@ -0,0 +1,43 @@
+import jwt from "jsonwebtoken";
+import tokenService from "./services/tokenService";
+import HTTPUtil from "./httpUtil";
+import { hash } from "./util";
+
+const httpUtil = new HTTPUtil();
+const {
+  ACCESS_TOKEN_SECRET,
+  ACCESS_TOKEN_LIFE,
+  REFRESH_TOKEN_SECRET,
+} = process.env;
+
+const verify = async (req, res, next) => {
+  try {
+    const { authorization } = req.headers;
+    if (!authorization) {
+      httpUtil.setError(403, "not authorized");
+      return httpUtil.send(res);
+    }
+    const token = authorization.split(" ")[1];
+    const oldAccessTokenId = hash(token);
+    jwt.verify(token, ACCESS_TOKEN_SECRET);
+    const refreshToken = await tokenService.getRefreshToken(oldAccessTokenId);
+    if (!refreshToken) {
+      httpUtil.setError(403, "not authorized");
+      return httpUtil.send(res);
+    }
+    jwt.verify(refreshToken, REFRESH_TOKEN_SECRET);
+    const newAccessToken = jwt.sign({}, ACCESS_TOKEN_SECRET, {
```

Comment:
new access token should be generated only when we call /refresh_tokens and not in verify middleware